lib/engine/step/g_1860/tracker.rb

```diff
@@ -0,0 +1,279 @@
+# frozen_string_literal: true
+
+require_relative '../base'
+
+module Engine
+  module Step
+    module G1860
+      module Tracker
+        def setup
+          @upgraded = false
+          @laid_track = 0
+          @laid_city = false
+        end
+
+        def can_lay_tile?(entity)
+          action = get_tile_lay(entity)
+          return false unless action
+
+          entity.tokens.any? && (@game.buying_power(entity) >= action[:cost]) && (action[:lay] || action[:upgrade])
```

Comment:
i don't think actions use symbols, i think they are all strings, double check this

Reply:
I didn't change this.

Reply:
This is a different "action" than the Action class. I didn't name it.